### PR TITLE
libckteec: Add RSA support

### PR DESCRIPTION
**Pull request by vesajaaskelainen**
These commits adds RSA support for:
- Key pair generation
- PKCS#1 v1.5 signing & verification support with several hash modes
- PSS signing & verification support
- OAEP encryption/decryption support

Relates to:
- https://github.com/OP-TEE/optee_os/issues/4283

Related PRs:
- https://github.com/OP-TEE/optee_os/pull/4742
- https://github.com/OP-TEE/optee_test/pull/534

### CODE REVIEW

File: libckteec/src/ck_debug.c [hunk 1]
```diff
@@ -125,6 +125,7 @@ const char *ckm2str(CK_MECHANISM_TYPE id)
 	CK2STR_ENTRY(CKM_RSA_PKCS);
 	CK2STR_ENTRY(CKM_RSA_9796);
 	CK2STR_ENTRY(CKM_RSA_X_509);
+	CK2STR_ENTRY(CKM_SHA1_RSA_PKCS);
```
Comment (ruchi393) [inline libckteec/src/ck_debug.c:128]: CKM_MD5_RSA_PKCS entry is missing

Comment (vesajaaskelainen) [reply libckteec/src/ck_debug.c:129]: Thanks for noticing. Added.

**Review by etienne-lms — COMMENTED** [review]
`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>`

File: libckteec/src/serialize_ck.c [hunk 2]
```diff
@@ -522,7 +522,14 @@ CK_RV serialize_ck_mecha_params(struct serializer *obj,
 	case CKM_ECDSA_SHA256:
 	case CKM_ECDSA_SHA384:
 	case CKM_ECDSA_SHA512:
+	case CKM_RSA_PKCS:
```
[the remark this answers is not in the document]
Comment (vesajaaskelainen) [reply libckteec/src/serialize_ck.c:592]: OK

**Comment by vesajaaskelainen** [comment]
Updated the PR based on comments.

Applied tags. Thanks.

**Comment by ruchi393** [comment]
`Reviewed-by: Ruchika Gupta <ruchika.gupta@linaro.org>`

**Comment by vesajaaskelainen** [comment]
Applied tags. Thanks for the reviews!

**Pull request merged**